group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorService.java

```diff
@@ -578,4 +619,19 @@ public void shutdown() {
         Utils.closeQuietly(runtime, "coordinator runtime");
         log.info("Shutdown complete.");
     }
+
+    private boolean isValidGroupId(String groupId, ApiKeys api) {
+        if (api == ApiKeys.OFFSET_COMMIT ||
+            api == ApiKeys.OFFSET_FETCH ||
+            api == ApiKeys.DESCRIBE_GROUPS ||
+            api == ApiKeys.DELETE_GROUPS
```

Comment:
i will take your suggestion for this PR. however, it does make more sense to have the logic in one place instead of using isGroupIdNotNull/isGroupIdNotEmpty based on the request.